examples/sn-dms-demo/src/store/wopi/reducers.ts

```diff
@@ -0,0 +1,22 @@
+import { ODataWopiResponse } from '@sensenet/client-core'
+import { AnyAction, combineReducers, Reducer } from 'redux'
+
+const wopiDataInitialState = {
+  accesstoken: '',
+  actionUrl: '',
+  expiration: 0,
+  faviconUrl: '',
+}
+
+export const wopiData: Reducer<ODataWopiResponse> = (state = wopiDataInitialState, action: AnyAction) => {
+  switch (action.type) {
+    case 'GET_WOPIDATA_SUCCESS':
```

Comment:
I don't see any action with this name. Where does this action come from?

Reply:
All the redux actions with ajax request are working this, the ones with the _SUCCES postfix are created in the middleware. :) If you want to catch when the response data is available, you have to use _SUCCESS actions in the reducer. :)

Reply:
got it! never realized this package `@sensenet/redux-promise-middleware` 😅